libraries/botbuilder-ai-orchestrator/src/orchestratorAdaptiveRecognizer.ts

```diff
@@ -110,228 +90,196 @@ export class OrchestratorAdaptiveRecognizer extends Recognizer implements Orches
+                    );
+                    if (ambiguousResults.length > 1) {
+                        const recognizerResults: Record<string, RecognizerResult> = ambiguousResults
+                            .map(
+                                (result): RecognizerResult => ({
+                                    text,
+                                    alteredText: result.closest_text,
+                                    entities: recognizerResult.entities,
+                                    intents: { [result.label.name]: { score: result.score } },
+                                })
+                            )
+                            .reduce((results: Record<string, RecognizerResult>, result: RecognizerResult) => {
+                                const guid = generate_guid();
+                                results[`${guid}`] = result;
+                                return results;
+                            }, {});
+
+                        // replace RecognizerResult with ChooseIntent => Ambiguous recognizerResults as candidates.
+                        recognizerResult = this.createChooseIntentResult(recognizerResults);
+                    }
+                }
             }
+        } else {
+            // Return 'None' if no intent matched.
+            recognizerResult.intents.None = { score: 1.0 };
         }
 
-        if (Object.entries(recognizerResult.intents).length == 0) {
-            recognizerResult.intents[this.noneIntent] = { score: 1.0 };
-        }
-
-        await dialogContext.context.sendTraceActivity(
-            'OrchestratorRecognizer',
+        await dc.context.sendTraceActivity(
+            'OrchestratorAdaptiveRecognizer',
             recognizerResult,
-            'RecognizerResult',
-            'Orchestrator recognizer RecognizerResult'
+            'OrchestratorAdaptiveRecognizer',
+            'Orchestrator Recognition'
+        );
+        this.trackRecognizerResult(
+            dc,
+            'OrchestratorAdaptiveRecognizer',
+            this.fillRecognizerResultTelemetryProperties(recognizerResult, telemetryProperties),
+            telemetryMetrics
         );
 
         return recognizerResult;
     }
 
-    /**
-     * @private
-     */
-    private AddTopScoringIntent(result: any, recognizerResult: RecognizerResult): void {
-        const topScoringIntent = result[0].label.name;
-        const topScore = result[0].score;
-
-        // if top scoring intent is less than threshold, return None
-        if (topScore < this.unknownIntentFilterScore) {
-            recognizerResult.intents['None'] = { score: 1.0 };
-        } else if (!recognizerResult.intents[topScoringIntent]) {
-            recognizerResult.intents[topScoringIntent] = {
-                score: topScore,
-            };
+    private _initializeModel() {
+        if (!this._modelPath) {
+            throw new Error(`Missing "ModelPath" information.`);
```

Comment:
Is there any reason not to perform this validation in the constructor?

Reply:
In a declarative based bot, the resource explorer always construct components with empty parameters, so the validation cannot be done in constructor.

Reply:
Makes sense!